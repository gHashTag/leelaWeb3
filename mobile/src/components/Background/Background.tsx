import React, { ReactNode } from 'react'

import { ScrollView, StyleSheet, View } from 'react-native'

import { W } from 'cons'
import { useGlobalBackground } from 'hooks'

interface BackgroundProps {
  children: ReactNode
  isScrollView?: boolean
  isFlatList?: boolean
}

const Background: React.FC<BackgroundProps> = ({
  children,
  isScrollView = false,
  isFlatList = false,
}) => {
  const backgroundStyle = useGlobalBackground()

  if (isScrollView) {
    return (
      <View style={[backgroundStyle, styles.flatlistStyle]}>
        <ScrollView
          contentContainerStyle={styles.scrollViewContent}
          showsVerticalScrollIndicator={false}
        >
          {children}
        </ScrollView>
      </View>
    )
  }

  if (isFlatList) {
    return (
      <View style={[backgroundStyle, styles.flatlistStyle]}>{children}</View>
    )
  }

  return <View style={[backgroundStyle, styles.container]}>{children}</View>
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  flatlistStyle: {
    alignItems: 'center',
  },
  scrollViewContent: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'flex-start',
    width: W,
  },
})

export { Background }
