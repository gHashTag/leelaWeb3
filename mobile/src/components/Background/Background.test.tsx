import React from 'react'

import { Text } from 'react-native'

import { render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import { Background } from './Background'

describe('Background Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Background>
          <Text testID="child-text">Hello, World!</Text>
        </Background>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders children', () => {
    const { getByTestId } = render(
      <Background>
        <Text testID="child-text">Hello, World!</Text>
      </Background>,
    )
    const childText = getByTestId('child-text')
    expect(childText).toBeTruthy()
    expect(childText.props.children).toBe('Hello, World!')
  })
})
