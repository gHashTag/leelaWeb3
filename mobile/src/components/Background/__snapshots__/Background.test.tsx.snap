// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Component renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "#F5F7F8",
      },
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    testID="child-text"
  >
    Hello, World!
  </Text>
</View>
`;
